Source/SwiftLintFramework/Rules/ConditionalReturnsOnNewline.swift

```diff
@@ -23,22 +23,43 @@ public struct ConditionalReturnsOnNewline: ConfigurationProviderRule, Rule, OptI
             "guard true else {\n return true\n}",
             "guard true,\n let x = true else {\n return true\n}",
             "if true else {\n return true\n}",
-            "if true,\n let x = true else {\n return true\n}"
+            "if true,\n let x = true else {\n return true\n}",
+            "if textField.returnKeyType == .Next {",
+            "if true { // return }",
+            "/*if true { */ return }",
         ],
         triggeringExamples: [
             "guard true else { return }",
             "if true { return }",
             "if true { break } else { return }",
+            "if true { break } else {       return }",
         ]
     )
 
     public func validateFile(file: File) -> [StyleViolation] {
         let pattern = "(guard|if)[^\n]*return[^\n]\n*"
         let excludingKinds = SyntaxKind.commentAndStringKinds()
-        return file.matchPattern(pattern, excludingSyntaxKinds: excludingKinds).map {
+        return file.rangesAndTokensMatching(pattern).filter { range, tokens in
+            let kinds = tokens.flatMap { SyntaxKind(rawValue: $0.type) }
+
+            guard kinds.filter(excludingKinds.contains).isEmpty else {
```

Comment:
This check caused false negative when match contains comment or string.
e.g. `"if true { return \"YES\" } else { return \"NO\" }"`.